plugins/urls/__init__.py

```diff
@@ -1,168 +1,209 @@
 # coding=utf-8
-"""
-A plugin to provide various URL-related tools and services.
-
-This plugin does quite a lot of things - URL title parsing, URL shortening,
-URL logging to database, specialized URL handlers.. Its partner plugin,
-URL-tools (in the contrib repo) extends its functionality using the API it
-exposes.
-"""
-
-__author__ = 'Gareth Coles'
-
-import fnmatch
 import re
-import socket
-import urlparse
-import urllib
-import urllib2
 
-from httplib import IncompleteRead
+from copy import copy
+from collections import defaultdict
 
-from bs4 import BeautifulSoup
 from kitchen.text.converters import to_unicode
-from netaddr import all_matching_cidrs
-
+from twisted.internet.defer import Deferred, inlineCallbacks, returnValue
+from twisted.python.failure import Failure
+from txrequests import Session
+from plugins.urls.lazy import LazyRequest
+from plugins.urls.priority import Priority
+from plugins.urls.shorteners.exceptions import ShortenerDown
 from system.command_manager import CommandManager
-from system.decorators.threads import run_async_threadpool
 from system.event_manager import EventManager
-from system.events.general import MessageReceived
-
-import system.plugin as plugin
-
 from system.protocols.generic.channel import Channel
-from system.protocols.generic.user import User
-
-from system.storage.formats import YAML, DBAPI
+from system.storage.formats import Formats
+from system.plugins.plugin import PluginObject
 from system.storage.manager import StorageManager
+from plugins.urls.constants import PREFIX_TRANSLATIONS
+from plugins.urls.events import URLsPluginLoaded
+from plugins.urls.handlers.website import WebsiteHandler
+from plugins.urls.matching import extract_urls
+from plugins.urls.shorteners.tinyurl import TinyURLShortener
+from plugins.urls.url import URL
+from utils.misc import str_to_regex_flags
 
-from .catcher import Catcher
+__author__ = 'Gareth Coles'
 
-from system.translations import Translations
-_ = Translations().get()
-__ = Translations().get_m()
 
+HTTP_S_REGEX = re.compile("http|https", flags=str_to_regex_flags("iu"))
 
-class URLsPlugin(plugin.PluginObject):
-    """
-    URLs plugin object
-    """
 
-    catcher = None
-    channels = None
+class URLsPlugin(PluginObject):
+    # Managers
     commands = None
```

Comment:
Don't need all this boilerplate anymore - `PluginObject` should have this stuff pre-set.